src/sync.rs

```diff
@@ -1,71 +1,11 @@
-//! Sync actors support
+//! Sync Actors support
 //!
-//! Sync actors could be used for cpu bound load. Only one sync actor
-//! runs within arbiter's thread. Sync actor process one message at a time.
-//! Sync arbiter can start multiple threads with separate instance of actor in
-//! each. Note on actor `stopping` lifecycle event, sync actor can not prevent
-//! stopping by returning `false` from `stopping` method.
-//! Multi consumer queue is used as a communication channel queue.
-//! To be able to start sync actor via `SyncArbiter`
-//! Actor has to use `SyncContext` as an execution context.
-//!
-//! ## Example
-//!
-//! ```rust
-//! # extern crate actix;
-//! # extern crate futures;
-//! use actix::prelude::*;
-//!
-//! struct Fibonacci(pub u32);
-//!
-//! impl Message for Fibonacci {
-//!     type Result = Result<u64, ()>;
-//! }
-//!
-//! struct SyncActor;
-//!
-//! impl Actor for SyncActor {
-//!     type Context = SyncContext<Self>;
-//! }
-//!
-//! impl Handler<Fibonacci> for SyncActor {
-//!     type Result = Result<u64, ()>;
-//!
-//!     fn handle(&mut self, msg: Fibonacci, _: &mut Self::Context) -> Self::Result {
-//!         if msg.0 == 0 {
-//!             Err(())
-//!         } else if msg.0 == 1 {
-//!             Ok(1)
-//!         } else {
-//!             let mut i = 0;
-//!             let mut sum = 0;
-//!             let mut last = 0;
-//!             let mut curr = 1;
-//!             while i < msg.0 - 1 {
-//!                 sum = last + curr;
-//!                 last = curr;
-//!                 curr = sum;
-//!                 i += 1;
-//!             }
-//!             Ok(sum)
-//!         }
-//!     }
-//! }
-//!
-//! fn main() {
-//!     System::run(|| {
-//!         // start sync arbiter with 2 threads
-//!         let addr = SyncArbiter::start(2, || SyncActor);
-//!
-//!         // send 5 messages
-//!         for n in 5..10 {
-//!             addr.do_send(Fibonacci(n));
-//!         }
-//!
-//! #       System::current().stop();
-//!     });
-//! }
-//! ```
+//! Sync Actors are actors that run multiple instances on a thread pool.
```

Comment:
You removed few details about lifetimes of sync actors. Please retain these details

Reply:
If there are important lifecycle details shouldn't these be mentioned on the start/stop/terminate functions them self then? I think that having them here adds confusion. This is the explanation of the high level concept of what a sync arbiter is, and then if there are lifecycle details, these should be explained on the relevant functions.

Said differently, in most cases does the normal usage case actually care about these lifecycles? IMO, they probably don't ....

Reply:
Even if most people don't care about lifecycle, it is should remain here for completeness sake.
Do not remove some details just because you think they are not important

Reply:
I'm not saying remove it, I'm saying put it in a more relevant section, where it has better context - not to mention it's current wording is confusing and dense. I certainly don't understand what it's trying to tell me :(. 

I think that for a general overview, we shouldn't add too many details, when here, we probably actually want to move the lifecycle part to sync context, which seems like a much more relevant location due to the presence of the stop/terminate/state methods. We could also add a link from SyncArbiter to the SyncContext docs to say "for lifetime of the actors in a SyncArbiter, see ...". I think that would be a better way to structure this documentation.

Reply:
It tells you that you're unable prevent sync actors from being stopped, once it is started (i.e. `stopping` method is called)

You can put it in separate paragraph as a Note